R/predict.sgdnet.R

```diff
@@ -129,6 +163,69 @@ lambda_interpolate <- function(lambda, s) {
   list(left = left, right = right, frac = sfrac)
 }
 
+#' Refit sgdnet object
+#'
+#' @param object an object of class `sgdnet`
+#' @param s a new lambda penalty (or several)
+#' @param ... new arguments to pass on to [sgdnet()]
+#'
+#' @return A new fit from [sgdnet()].
+#' @keywords internal
+refit <- function(object, s, ...) {
+  lambda <- object$lambda
+  which <- match(s, lambda, FALSE)
+
+  if (!all(which > 0)) {
+    lambda <- unique(rev(sort(c(s, lambda))))
+    object <- stats::update(object, lambda = lambda, ...)
+  }
+}
+
+
+#' Bind intercept with coefficients
+#'
+#' @param beta coefficients
+#' @param a0 intercept
+#'
+#' @return A matrix (or listof matrices).
+#' @keywords internal
+bind_intercept <- function(beta, a0) {
+  if (is.list(beta)) {
+    for (i in seq_along(beta))
+      beta[[i]] <- bind_intercept(beta[[i]], a0[i, , drop = FALSE])
+    beta
+  } else {
+    out <- methods::rbind2(a0, beta)
+    rownames(out)[1] <- "(Intercept)"
+    out
+  }
+}
+
+#' Interpolate coefficients
+#'
+#' @param beta coeffieicnts
+#' @param s lambda penalty
+#' @param lamlist a list of lambda interpolation parameters, returned from
+#'   [lambda_interpolate()].
+#'
+#' @return A matrix (or list of matrices) with new coefficients based
+#'   on linearly interpolating from new and old lambda values.
+#' @keywords internal
```

Comment:
I think you're right... although I do sometimes think it is nice to have the documentation included but not visible because it means I can do ?function() whenever I find an internal function in the code that I don't remember the purpose of, but perhaps it is cleaner to add @noRd anyway.